### PR TITLE
Add note about paths with %

### LINKED ISSUE #16066: Default compilation includes does not work if your path has '%'
We had a git repo with a space, that translated into a % and then this was not working. It took us quite some time to be able to troubleshoot. You should include a warning in this section:
https://docs.microsoft.com/en-us/dotnet/core/tools/csproj#default-compilation-includes-in-net-core-projects

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: aa484235-d872-f50e-96fc-aa5b4a2a42fa
* Version Independent ID: 871c22e0-65a2-f3a4-41b9-f050489dba7a
* Content: [.NET project SDK overview](https://docs.microsoft.com/en-us/dotnet/core/project-sdk/overview#default-includes-and-excludes)
* Content Source: [docs/core/project-sdk/overview.md](https://github.com/dotnet/docs/blob/master/docs/core/project-sdk/overview.md)
* Product: **dotnet-fundamentals**
* GitHub Login: @gewarren
* Microsoft Alias: **gewarren**

### PR DESCRIPTION
**Pull request by gewarren**
Fixes #16066.

### CODE REVIEW

**Review by tdykstra — APPROVED** [review]
This might be more discoverable if it were created as a NETSDK1004.md file in the core/tools/sdk-errors folder, following the pattern of the other docs such as https://docs.microsoft.com/en-us/dotnet/core/tools/sdk-errors/netsdk1005

**Comment by gewarren** [comment]
Thanks Tom @tdykstra . Can you take another look? I created a netsdk1004 page. My only concern is that it's not very useful because it only provides one reason you might see the error.

https://review.docs.microsoft.com/en-us/dotnet/core/tools/dotnet-build?branch=gewarren-patch-3-1#feedback

**Comment by tdykstra** [comment]
>My only concern is that it's not very useful because it only provides one reason you might see the error.

I think it's fine for now, it's a base we can add to if and when other causes for the same message turn up.

I wonder about the value of having an alert box in the dotnet build command article, though, where it will draw attention of 100% of doc users although it applies to extremely few of them.  I think you could eliminate it from that doc and only document it in the error page.

File: docs/core/tools/sdk-errors/netsdk1004.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+---
+# NETSDK1004: Assets file not found
+
+**This article applies to:** ✔️ .NET Core 2.1.100 SDK and later versions
+
+This error occurs when the assets file *project.assets.json* is not found during build. The full error message is similar to:
+
+**NETSDK1004: Assets file 'C:\IncorrectPath\project.assets.json' not found. Run a NuGet package restore to generate this file.**
+
+One reason you might encounter warning NETSDK1004 is if you run the `dotnet build` command from a directory path that contains a `%` character.
```
Comment (marcpopMSFT) [inline docs/core/tools/sdk-errors/netsdk1004.md:17]: We've seen an increase in this error recently from two additional causes:
1. A change to the project file that wasn't automatically detected and restored by the project system
2. A project being restored separately by an older version of Nuget.exe

For either, the customer should open a command line and run `dotnet restore` on the project in question.

**Pull request merged**